internal/pkg/slicer/closer/closer.go

```diff
@@ -1,6 +1,32 @@
 // Package closer provides utility to close readers or writers chain in LIFO order.
 package closer
 
+// Closers type is a list of close callbacks, which are invoked in LIFO order.
+//
+// # Closing Writers Chain
+//
+// Writer Close method can have the following uses:
+//   - releasing resources
+//   - flushing internal buffers
+//   - write stream tail
+//
+// For reasons above, it is necessary to call close methods in the correct order, see example bellow:
+//
+//	WRITE        >>> BufferIn -> GzipWriter -> BufferOut -> os.File
+//	OPEN                4     <-     3      <-     2     <-    1            (opening the file is the FIRST step)
+//	CLOSE/FLUSH         1     ->     2      ->     3     ->    4            (closing the file is the LAST step)
+//
+// # Closing Readers Chain
+//
+// Reader Close method is used only to release resources.
+//
+// The stream ends with an EndOfFile (io.EOF) error, so there is no risk that we have not read everything.
+//
+// Therefore, we can close the readers in any order, so we use the same LIFO order.
+//
+//	READ io.EOF  <<< BufferOut <- GzipReader <- BufferIn <- os.File
+//	OPEN                4      <-     3      <-    2     <-    1            (opening the file is the first step)
+//	CLOSE              4/1     <->   3/2     <->  2/3    <->  1/4           (closing the file is the FIRST or the LAST step)
 type Closers []func() error
```

Comment:
`closer` package was created to facilitate the work of closing writers chains, therefore it uses LIFO order (same as the `defer` statement), see [closer.Closers.Close](https://github.com/keboola/processor-split-table/blob/3e61fabf16a65546841ae4129f116cbd62126e5d/internal/pkg/slicer/closer/closer.go#L37-L46) method.